### ISSUE TITLE
Is it possible to see SQL Agent jobs in an outlook-style calendar view?

### ISSUE THREAD

**Issue by tdinh2011** [trimble-oss/dba-dash#479]
Create an Outlook-Style calendar view, color coded by job types and has the following info: 
- server name, job category/name, last run, next run, duration

**Comment by HansLindgren** [comment]
This would be a really cool feature!

**Comment by DavidWiseman** [comment]
WSUHoey on slack suggested Google charts similar to what [dba tools](https://dbatools.io/timeline/) uses.  A .NET control would be better but the options for a free open-source control that can be used in DBA Dash seem to be limited.  There are some old projects that appear to be dead. 

Google charts might be the best option.  It's at least technically possible:
![image](https://user-images.githubusercontent.com/33157668/208188606-7fb753b1-996f-47cc-bd19-330f6844ee6f.png)

The above was a quick PoC using a WebView2 control to load the chart data from one the examples.

```c#
namespace WinFormsApp1
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
           

        }
        Microsoft.Web.WebView2.WinForms.WebView2 web1 = new();
        private async void Form1_Load(object sender, EventArgs e)
        {
            Controls.Add(web1);
            web1.Dock= DockStyle.Fill;
            await web1.EnsureCoreWebView2Async(null);
            web1.NavigateToString(@"
<html>
<head>
<script type=""text/javascript"" src=""https://www.gstatic.com/charts/loader.js""></script>

<script type=""text/javascript"">
  google.charts.load(""current"", {packages:[""timeline""]});
  google.charts.setOnLoadCallback(drawChart);
  function drawChart() {

    var container = document.getElementById('example3.1');
    var chart = new google.visualization.Timeline(container);
    var dataTable = new google.visualization.DataTable();
    dataTable.addColumn({ type: 'string', id: 'Position' });
    dataTable.addColumn({ type: 'string', id: 'Name' });
    dataTable.addColumn({ type: 'date', id: 'Start' });
    dataTable.addColumn({ type: 'date', id: 'End' });
    dataTable.addRows([
      [ 'President', 'George Washington', new Date(1789, 3, 30), new Date(1797, 2, 4) ],
      [ 'President', 'John Adams', new Date(1797, 2, 4), new Date(1801, 2, 4) ],
      [ 'President', 'Thomas Jefferson', new Date(1801, 2, 4), new Date(1809, 2, 4) ],
      [ 'Vice President', 'John Adams', new Date(1789, 3, 21), new Date(1797, 2, 4)],
      [ 'Vice President', 'Thomas Jefferson', new Date(1797, 2, 4), new Date(1801, 2, 4)],
      [ 'Vice President', 'Aaron Burr', new Date(1801, 2, 4), new Date(1805, 2, 4)],
      [ 'Vice President', 'George Clinton', new Date(1805, 2, 4), new Date(1812, 3, 20)],
      [ 'Secretary of State', 'John Jay', new Date(1789, 8, 25), new Date(1790, 2, 22)],
      [ 'Secretary of State', 'Thomas Jefferson', new Date(1790, 2, 22), new Date(1793, 11, 31)],
      [ 'Secretary of State', 'Edmund Randolph', new Date(1794, 0, 2), new Date(1795, 7, 20)],
      [ 'Secretary of State', 'Timothy Pickering', new Date(1795, 7, 20), new Date(1800, 4, 12)],
      [ 'Secretary of State', 'Charles Lee', new Date(1800, 4, 13), new Date(1800, 5, 5)],
      [ 'Secretary of State', 'John Marshall', new Date(1800, 5, 13), new Date(1801, 2, 4)],
      [ 'Secretary of State', 'Levi Lincoln', new Date(1801, 2, 5), new Date(1801, 4, 1)],
      [ 'Secretary of State', 'James Madison', new Date(1801, 4, 2), new Date(1809, 2, 3)]
    ]);

    chart.draw(dataTable);
  }
</script>
</head>

<div id=""example3.1"" style=""height: 200px;""></div>

</html>
");
        }
    }
}
```

The javascript that generates the chart is pulled from the internet so this wouldn't work on a computer without internet connectivity - might not be a big deal.

Google charts seem easy to work with and there are some[ good examples](https://developers.google.com/chart/interactive/docs/gallery/timeline) provided.

JobHistory is captured by DBA Dash but the default retention is only 8 days (probably enough).  Job stats are kept for longer aggregated into 60min periods.  

I'm guessing the main purpose of the timeline would be to identify overlapping jobs and assist with picking job schedules.  To get a better idea of what jobs are running at what parts of the day.  If there are any other use cases, let me know.

**Comment by DavidWiseman** [comment]
WebView2 Runtime is required or a file load exception will occur:
https://developer.microsoft.com/en-us/microsoft-edge/webview2/

A check will need to be added for this.

**Comment by DavidWiseman** [comment]
The timeline view is available in 2.31.0. 🚀

Note: The color coding is done based on the success/failure of the job.  It's based off the history of execution and doesn't include last run/next run time.  You can see last run anyway depending on the date range selected.  Category filters are available.
